register

Thanachot-W opened the pull request.

Thanachot-W committed:
register

FC4RICA commented on app/src/main/java/com/fc4rica/bonbaan/ui/auth/RegisterScreen.kt:
```diff
@@ -2,8 +2,31 @@ package com.fc4rica.bonbaan.ui.auth
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.example.bonbaan.ui.components.ButtonComponent
+import com.example.bonbaan.ui.components.TextFieldComponent
```
Please change import to match this project. It's com.fc4rica.bonbaan

FC4RICA commented on app/src/main/java/com/fc4rica/bonbaan/ui/auth/RegisterScreen.kt:
Can you please add more TextFields: username, firstname, lastname.

FC4RICA reviewed (changes requested):
Good work. But there's a few things that need to change. Can you move the Button.kt and TextField.kt into `app/src/main/java/com/fc4rica/bonbaan/ui/components`? That would be great!

FC4RICA committed:
Refactor: Introduce BonBaanButton and update RegisterScreen

- Added `BonBaanButton` composable for customizable button styles.
- Renamed and moved `ButtonComponent` to `BonBaanButton` and `component` to `com.fc4rica.bonbaan.ui.components`.
- Implemented various `ButtonVariant` styles: PRIMARY, SECONDARY, OUTLINED, TEXT, and DESTRUCTIVE.
- Updated `RegisterScreen` to use `BonBaanButton`.
- Added `Preview` for `BonBaanButton` and `RegisterScreen`.
- Corrected the package of `Button` and `TextField` component.
- Corrected start destination of authenticationGraph.

FC4RICA committed:
Refactor: Update RegisterScreen UI and add BonBaanTextField

- Added `BonBaanTextField` and updated `RegisterScreen` to utilize it.
- Updated `Button` component, adding `TEXT` variant.
- Updated `RegisterScreen` layout and content, with improved text elements.
- Replaced `TextFieldComponent` with `BonBaanTextField` in `RegisterScreen`.
- Added password visibility toggle to `BonBaanTextField`.
- Updated navigation in `RegisterScreen` to navigate to `LoginScreen`.
- Added `ContentPadding` to `BonBaanButton`.

FC4RICA committed:
Feat: Add logo to RegisterScreen and update UI

- Added an app logo to the `RegisterScreen` using the `Image` composable and the `logo1` drawable resource.
- Updated the UI of the `RegisterScreen` to include the logo, adjust spacing, and correct spelling:
  - Added an `Image` composable to display the `logo1` with specific dimensions and padding.
  - Introduced `Spacer` composables for vertical spacing between elements.
  - Corrected the spelling of "อีเมล์" to "อีเมล" in the `BonBaanTextField` label and the body text.
- Added logo1.png file.

FC4RICA committed:
Refactor: Update RegisterScreen navigation and button actions

- Removed navigation from the "Next" button in `RegisterScreen`.
- Added navigation from the "Login" button in `RegisterScreen` to the `Screen.Login` route.
